core/templates/pages/exploration-player-page/learner-experience/ratings-and-recommendations.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for ratings and recommendations component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { EventEmitter, NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, fakeAsync, TestBed, tick, waitForAsync } from '@angular/core/testing';
+import { NgbPopoverModule } from '@ng-bootstrap/ng-bootstrap';
+import { AlertsService } from 'services/alerts.service';
+import { UrlService } from 'services/contextual/url.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { UserService } from 'services/user.service';
+import { LearnerViewRatingService } from '../services/learner-view-rating.service';
+import { MockLimitToPipe } from '../templates/information-card-modal.component.spec';
+import { RatingsAndRecommendationsComponent } from './ratings-and-recommendations.component';
+
+describe('Ratings and recommendations component', () => {
+  let fixture: ComponentFixture<RatingsAndRecommendationsComponent>;
+  let componentInstance: RatingsAndRecommendationsComponent;
+  let alertsService: AlertsService;
+  let learnerViewRatingService: LearnerViewRatingService;
+  let urlService: UrlService;
+  let userService: UserService;
+
+  const mockNgbPopover = jasmine.createSpyObj(
+    'NgbPopover', ['close', 'toggle']);
+
+
+  class MockWindowRef {
+    nativeWindow = {
+      location: {
+        search: '',
+        pathname: '/path/name',
+        reload: () => {}
+      }
+    };
+  }
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        NgbPopoverModule
+      ],
+      declarations: [
+        RatingsAndRecommendationsComponent,
+        MockLimitToPipe
+      ],
+      providers: [
+        AlertsService,
+        LearnerViewRatingService,
+        UrlService,
+        UserService,
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(RatingsAndRecommendationsComponent);
+    componentInstance = fixture.componentInstance;
+    alertsService = TestBed.inject(AlertsService);
+    learnerViewRatingService = TestBed.inject(LearnerViewRatingService);
+    urlService = TestBed.inject(UrlService);
+    userService = TestBed.inject(UserService);
+  });
+
+  it('should initialize', fakeAsync(() => {
```

Comment:
Probably should be more like "should do X on initialize", e.g. "populate internal properties"

Reply:
Done